src/chainstate/stacks/miner.rs

```diff
@@ -3994,6 +4006,131 @@ pub mod test {
         }
     }
 
+    #[test]
+    fn test_build_anchored_blocks_bad_nonces() {
+        let mut privks = vec![];
+        let mut balances = vec![];
+        let num_blocks = 10;
+
+        for _ in 0..num_blocks {
+            let privk = StacksPrivateKey::new();
+            let addr = StacksAddress::from_public_keys(C32_ADDRESS_VERSION_TESTNET_SINGLESIG, &AddressHashMode::SerializeP2PKH, 1, &vec![StacksPublicKey::from_private(&privk)]).unwrap();
+
+            privks.push(privk);
+            balances.push((addr.to_account_principal(), 100000000));
+        }
+
+        let mut peer_config = TestPeerConfig::new("test_build_anchored_blocks_too_expensive_transactions", 2013, 2014);
```

Comment:
The values `2013` and `2014` are port numbers, which will be bound in the subsequent call to `TestPeer::new()` below.  I haven't checked myself, but please make sure they're not used elsewhere in this file, lest this test fail due to a port-in-use error.

Reply:
before I made the change, the ci was failing while running that single test worked locally, so these values are probably ok.